src/core/meshcop/joiner.cpp

```diff
@@ -450,11 +450,16 @@ otError Joiner::PrepareJoinerFinalizeMessage(const char *aProvisioningUrl,
 
 void Joiner::FreeJoinerFinalizeMessage(void)
 {
+    VerifyOrExit(mState == OT_JOINER_STATE_IDLE);
```

Comment:
nit: include the NULL check in this `VerifyOrExit`?

Reply:
made it in separate `VerifyOrExit` to make it clearer.